lxd/storage/drivers/interface.go

```diff
@@ -35,14 +35,15 @@ type Driver interface {
 	CreateVolumeFromCopy(vol Volume, srcVol Volume, copySnapshots bool, op *operations.Operation) error
 	DeleteVolume(volType VolumeType, volName string, op *operations.Operation) error
 	RenameVolume(volType VolumeType, volName string, newName string, op *operations.Operation) error
+	UpdateVolume(vol Volume, changedKeys map[string]struct{}) error
```

Comment:
Why the weird map instead of a []string?

Reply:
Well, I supposed we actually want map[string]string so we get access to the value.
It may not be needed for the `dir` backend, but it's going to be needed for some of the other drivers where a volume config change actually needs to result in a filesystem action.

Reply:
I think I would expect `vol.Config` to contain the old config with `newConfig map[string]string` giving UpdateVolume access to the new config.

Reply:
@stgraber it seemed like it better aligned with the operations that were performed on it, i.e mark a key as changed (even though it could be found to be changed in multiple steps), and therefore when a slice duplicate key detection was needed before it could be added, whereas using a map allows the key to just be set without checking if its already sets. The `struct{}` part indicates the map item has no value itself.